### ISSUE TITLE
Executor._error_cycle_task fails due to BespokeOptimizationSchema having no "status" attribute

### ISSUE THREAD

**Issue by lilyminium** [openforcefield/bespoke-fit#31]
The method fails here:
https://github.com/openforcefield/bespoke-fit/blob/7f02c8363e91e05bc404309ce58a678ad4eacc21/openff/bespokefit/bespoke/executor.py#L350

However, in the code above, `task.status` is only assigned under specific conditions:
https://github.com/openforcefield/bespoke-fit/blob/7f02c8363e91e05bc404309ce58a678ad4eacc21/openff/bespokefit/bespoke/executor.py#L311-L329

I'm still trying to figure bespokefit and the new API out but I'm guessing you may want to give BespokeOptimizationSchema a placeholder `status` attribute (=Status.Undefined) or throw in a `if hasattr(task, "status")`?

Edit: error trace here:

```
Process Process-1:
Traceback (most recent call last):
  File "/Users/lily/anaconda3/envs/bespokefit/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/Users/lily/anaconda3/envs/bespokefit/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/lily/pydev/bespoke-fit/openff/bespokefit/bespoke/executor.py", line 381, in error_cycle
    self._error_cycle_task(task=task, client=client)
  File "/Users/lily/pydev/bespoke-fit/openff/bespokefit/bespoke/executor.py", line 350, in _error_cycle_task
    elif task.status == Status.CollectionError:
AttributeError: 'BespokeOptimizationSchema' object has no attribute 'status'
```

**Comment by SimonBoothroyd** [comment]
Thanks for reporting this @lilyminium - this will likely be fixed by #54 and the associated future PRs that go along with the refactor.

**Comment by SimonBoothroyd** [comment]
I'll go ahead and close this now that the new executor is in place. Please open an issue if you have any problems with the new approach!
